Source/DafnyLanguageServer/Plugins/PluginConfiguration.cs

```diff
@@ -0,0 +1,25 @@
+﻿// SPDX-License-Identifier: MIT
+using System;
+using System.Collections.Generic;
+
+namespace Microsoft.Dafny.LanguageServer.Plugins;
+
+/// <summary>
+/// Plugins should define a class that extends PluginConfiguration,
+/// in order to receive plugin-specific command-line arguments by overwriting the method `ParseArguments`
+/// It is also used to provide to Dafny a list of Rewriter instances using the method `GetRewriters`, and Compiler
+/// instances using `GetCompilers`
+///
+/// If the plugin defines no PluginConfiguration, then Dafny will instantiate every sub-class
+/// of Rewriter and Compiler from the plugin (providing rewriters with an ErrorReporter in the constructor
+/// as the first and only argument, and compilers with no arguments).
+/// </summary>
+public abstract class PluginConfiguration : Dafny.Plugins.PluginConfiguration {
+  /// <summary>
+  /// Override this method to provide quick fixers
+  /// </summary>
+  /// <returns>An array of quick fixers implemented by this plugin</returns>
+  public virtual DafnyCodeActionProvider[] GetDafnyCodeActionProviders() {
```

Comment:
Given that you changed `DafnyCodeActionProvider` to return an `IEnumerable`, would it make sense to do so here, too? If you don't do that, I think you could instead remove `using System.Collections.Generic` above.

Reply:
Good suggestion. For consistency with the original `Dafny.Plugins.PluginConfiguration` that returns `Compiler[]` and `Rewriter[]`, I prefer to keep returning an array here. So I removed `using System.Collections.Generic`